Libraries/LibRegex/RegexMatch.h

```diff
@@ -121,8 +121,11 @@ class RegexStringView {
                 return RegexStringView { T { *optional_string_storage } };
             },
             [&](Utf16View) {
-                optional_utf16_storage = AK::utf32_to_utf16(Utf32View { data.data(), data.size() }).release_value_but_fixme_should_propagate_errors();
-                return RegexStringView { Utf16View { optional_utf16_storage } };
+                auto conversion_result = utf32_to_utf16(Utf32View { data.data(), data.size() }).release_value_but_fixme_should_propagate_errors();
+                optional_utf16_storage = conversion_result.data;
+                auto view = Utf16View { optional_utf16_storage };
+                view.unsafe_set_code_point_length(conversion_result.code_point_count);
+                return RegexStringView { view };
```

Comment:
UAF or what. who owns the new string?

Reply:
oh I see. ya know, if you captured optional_utf16_storage by reference explicitly I wouldn't have questioned it.

Reply:
sure, will feex that too